src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -94,6 +95,52 @@ public void ThrowOnErrors()
         /// <returns></returns>
         public FormulaType GetNodeType(TexlNode node) => FormulaType.Build(_binding.GetType(node));
 
+        /// <summary>
+        /// Checks whether a call to a function with name <paramref name="fncName" /> is valid with argument list
+        /// <paramref name="args" />. Additionally returns (as an out parameter) the return type of this invocation.
+        /// 
+        /// Note: all arguments must belong to the formula that belongs to this <see cref="CheckResult" />.
+        /// </summary>
+        /// <param name="fncName"></param>
+        /// <param name="args"></param>
+        /// <param name="retType"></param>
+        /// <returns></returns>
+        public bool ValidateInvocation(string fncName, IReadOnlyList<TexlNode> args, out FormulaType retType)
+        {
+            retType = null;
+
+            if (args is null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            foreach (var arg in args)
+            {
+                if (arg is null)
+                {
+                    throw new ArgumentNullException(nameof(args));
+                }
+
+                if (!_binding.IsNodeValid(arg))
+                {
+                    throw new ArgumentException("Argument does not belong to this result");
+                }
+            }
+
+            var fnc = _binding.NameResolver.Functions
+                              .Where(fnc => fnc.Name == fncName && args.Count >= fnc.MinArity && args.Count <= fnc.MaxArity)
+                              .FirstOrDefault();

```

Comment:
>.FirstOrDefault(); [](http://example.com/codeflow?start=30&length=18)

There could be multiple matches (like overloads like Sum function). We could just loop.